src/main/java/frc/robot/RobotContainer.java

```diff
@@ -33,114 +35,121 @@
+        // OperatorConstants.LEFT_X_DEADBAND),
+        // () -> driverXbox.getRawAxis(2));
+
+        // AbsoluteDriveAdv closedAbsoluteDriveAdv = new
+        // AbsoluteDriveAdv(SwerveSubsystem.getInstance(),
+        // () -> MathUtil.applyDeadband(driverXbox.getLeftY(),
+        // OperatorConstants.LEFT_Y_DEADBAND),
+        // () -> MathUtil.applyDeadband(driverXbox.getLeftX(),
+        // OperatorConstants.LEFT_X_DEADBAND),
+        // () -> MathUtil.applyDeadband(driverXbox.getRightX(),
+        // OperatorConstants.RIGHT_X_DEADBAND),
+        // driverXbox::getYButtonPressed,
+        // driverXbox::getAButtonPressed,
+        // driverXbox::getXButtonPressed,
+        // driverXbox::getBButtonPressed);
+
+        // TeleopDrive simClosedFieldRel = new
+        // TeleopDrive(SwerveSubsystem.getInstance(),
+        // () -> MathUtil.applyDeadband(driverXbox.getLeftY(),
+        // OperatorConstants.LEFT_Y_DEADBAND),
+        // () -> MathUtil.applyDeadband(driverXbox.getLeftX(),
+        // OperatorConstants.LEFT_X_DEADBAND),
+        // () -> driverXbox.getRawAxis(2), () -> true);
+        TeleopDrive closedFieldRel = new TeleopDrive(
+                SwerveSubsystem.getInstance(),
+                () -> MathUtil.applyDeadband(driverController.getRawAxis(1),
+                        OperatorConstants.LEFT_Y_DEADBAND),
+                () -> MathUtil.applyDeadband(driverController.getRawAxis(0),
+                        OperatorConstants.LEFT_X_DEADBAND),
+                () -> -driverController.getRawAxis(4), () -> true);
+
+        SwerveSubsystem.getInstance().setDefaultCommand(
+                !RobotBase.isSimulation() ? closedFieldRel : closedFieldRel);
+        autoChooser = AutoBuilder.buildAutoChooser();
+        SmartDashboard.putData("Auto Chooser", autoChooser);
+    }
+
+    /**
+     * Use this method to define your trigger->command mappings. Triggers can be
+     * created via the
+     * {@link Trigger#Trigger(java.util.function.BooleanSupplier)} constructor with
+     * an arbitrary predicate, or via the
+     * named factories in
+     * {@link edu.wpi.first.wpilibj2.command.button.CommandGenericHID}'s subclasses
+     * for
+     * {@link CommandXboxController
+     * Xbox}/{@link edu.wpi.first.wpilibj2.command.button.CommandPS4Controller PS4}
+     * controllers or {@link edu.wpi.first.wpilibj2.command.button.CommandJoystick
+     * Flight joysticks}.
+     */
+    private void configureBindings() {
+        // Schedule `ExampleCommand` when `exampleCondition` changes to `true`
+
+        driverController.button(1).onTrue((new InstantCommand(SwerveSubsystem.getInstance()::zeroGyro)));
+        driverController.button(2).onTrue((new InstantCommand(() -> {
+            SmartDashboard.putNumber("button press", 0);
+            // Not safe type casting, could break but should be obvious
+            NavXSwerve navx = (NavXSwerve) SwerveSubsystem.getInstance().getSwerveDrive().imu;
```

Comment:
Why is this not safe type casting? Can `imu` be `null`?

Reply:
This is from the original code that is already on the Main branch.

Reply:
The cast is not safe because `imu` can _potentially_ be some other Swerve type, not exactly NavXSwerve, but in reality it should be NavXSwerve. The cast is not clean as we are forcing (downcasting) from a parent class to an inherited class.